irohad/main/peers_file_reader.hpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_PEERS_FILE_READER_HPP
+#define IROHA_PEERS_FILE_READER_HPP
+
+#include <boost/optional.hpp>
+#include "interfaces/common_objects/common_objects_factory.hpp"
+#include "interfaces/common_objects/peer.hpp"
+
+namespace iroha {
+  namespace main {
+    /**
+     * Peers reader interface from a file
+     */
+    class PeersFileReader {
+     public:
+      /**
+       * Opens file with peers and returns its contents
+       * @param name - file name to open
+       * @return optional for a file contents or boost::none
+       */
+      virtual boost::optional<std::string> openFile(
```

Comment:
quite a strange interface. why not initialize factories in instance constructor, and provide a single method to get an optional peers vector by file path at `PeersFileReader` interface?

Reply:
by the way, `expected::Result<PeersList, std::string>` seems a more suitable return type for `readPeers`. It can provide a human-readable error reason (no such file, no read access, format violation, ...).